source/class.cpp

```diff
@@ -1172,12 +1184,23 @@ void ClassBinder::bind(Context &context)
 	string const trampoline_name = callback_structure_constructible ? callback_structure_name(C) : "";
 	string const binding_qualified_name = callback_structure_constructible ? callback_structure_name(C) : qualified_name;
 
-	string maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name); // for now enable std::shared_ptr by default
-	if( is_inherited_from_enable_shared_from_this(C) ) maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name);
+	string shared_ptr = "std::shared_ptr";
+	bool use_custom_shared = Config::get().use_custom_shared();
+	if( use_custom_shared )
+		shared_ptr = Config::get().custom_shared();
```

Comment:
It is second time we calculate this, perhaps adding dedicated function would be best to avoid code duplication? It might be best to put this into config.cpp 🤔
```C++
std::string shared_ptr_class_name() {
  return shared_ptr = Config::get().use_custom_shared() ? Config::get().custom_shared() : "std::shared_ptr"; 
}
```